Makefile

```
@@ -2,15 +2,66 @@ ORG ?= spacemeshos
 IMAGE ?= poet
 BINARY := poet
 
+all: install build
+.PHONY: all
 
-ifdef TRAVIS_BRANCH
-        BRANCH := $(TRAVIS_BRANCH)
-else
-        BRANCH := $(shell git rev-parse --abbrev-ref HEAD)
-endif
+test:
+	gotestsum -- -timeout 5m -p 1 ./...
+.PHONY: test
 
-all: build
-.PHONY: all
+install:
+	go mod download
+	curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s v1.50.0
+	go install github.com/spacemeshos/go-scale/scalegen@v1.0.0
+	go install gotest.tools/gotestsum@v1.8.2
+	go install honnef.co/go/tools/cmd/staticcheck@latest
+.PHONY: install
+
+tidy:
+	go mod tidy
+.PHONY: tidy
+
+test-tidy:
+	# Working directory must be clean, or this test would be destructive
+	git diff --quiet || (echo "\033[0;31mWorking directory not clean!\033[0m" && git --no-pager diff && exit 1)
+	# We expect `go mod tidy` not to change anything, the test should fail otherwise
+	make tidy
+	git diff --exit-code || (git --no-pager diff && git checkout . && exit 1)
+.PHONY: test-tidy
+
+test-fmt:
+	git diff --quiet || (echo "\033[0;31mWorking directory not clean!\033[0m" && git --no-pager diff && exit 1)
+	# We expect `go fmt` not to change anything, the test should fail otherwise
+	go fmt ./...
+	git diff --exit-code || (git --no-pager diff && git checkout . && exit 1)
+.PHONY: test-fmt
```

I keep mistaking these targets with "tidying tests" and "formatting tests". How about renaming these targets to:
```suggestion
check-tidy:
	# Working directory must be clean, or this test would be destructive
	git diff --quiet || (echo "\033[0;31mWorking directory not clean!\033[0m" && git --no-pager diff && exit 1)
	# We expect `go mod tidy` not to change anything, the test should fail otherwise
	make tidy
	git diff --exit-code || (git --no-pager diff && git checkout . && exit 1)
.PHONY: check-tidy

check-fmt:
	git diff --quiet || (echo "\033[0;31mWorking directory not clean!\033[0m" && git --no-pager diff && exit 1)
	# We expect `go fmt` not to change anything, the test should fail otherwise
	go fmt ./...
	git diff --exit-code || (git --no-pager diff && git checkout . && exit 1)
.PHONY: check-fmt
```

---

I went with consistency since these targets are also named `test-fmt` and `test-tidy` in other repositories. I'm OK with check instead but then we should also update it for other components of spacemesh

---

For now I leave them as they are for consistency; we can do a rename for all repositories in the future 🙂